### ISSUE TITLE
Representation of a trained model? 

### ISSUE THREAD

**Issue by erdalkaraca** [neuromorphs/NIR#115]
Not sure if I miss something, but it seems to me that NIR is not considered to also represent a trained model, just its architecture, i.e. no model coefficients (and maybe hyper params).

**Comment by bvogginger** [comment]
@erdalkaraca A NIR model represents both the architecture and the model parameters (weights, biases, neuron thresholds ...). This can be either a trained SNN or just a manually created model with randomized or customized weights.

Yet, a NIR model does not represent a model checkpoint. Hence, it does not contain any training-related parameters such as learning rate, epoch, gradients etc.

If you want to store any training details in the NIR graph, you could use the `metadata` attribute of the NIRGraph (https://github.com/neuromorphs/NIR/blob/main/nir/ir/graph.py#L32C5-L32C13), which is a standard python dictionary. Each NIR node can have its own metadata.

If you have further questions please feel free to ask back or join the discord channel (https://discord.gg/JRMRGP9h3c).

Bernhard
